cql3/statements/select_statement.cc

```diff
@@ -274,13 +277,16 @@ select_statement::make_partition_slice(const query_options& options) const
     auto bounds =_restrictions->get_clustering_bounds(options);
     if (bounds.size() > 1) {
         auto comparer = position_in_partition::less_compare(*_schema);
+
         auto bounds_sorter = [&comparer] (const query::clustering_range& lhs, const query::clustering_range& rhs) {
             return comparer(position_in_partition_view::for_range_start(lhs), position_in_partition_view::for_range_start(rhs));
         };
         std::sort(bounds.begin(), bounds.end(), bounds_sorter);
     }
     if (_is_reversed) {
         std::reverse(bounds.begin(), bounds.end());
+        for (auto& bound : bounds)
+            bound = query::reverse(bound);
```

Comment:
Missing curly braces

Reply:
This patch is too difficult for me to review. It touches too many places, so I can't tell if somewhere is missed or if some place does something wrong.

One approach is to use a wave to make those changes. Each patch adjusts a callee and its callers, or a caller and its callees, leaving the rest of the system untouched. Eventually the wave propagates over the whole code base. In fact that's what we used during the original native reversed queries transformation.